src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -87,23 +89,46 @@ void DeleteInteractionChoiceSetRequest::Run() {
     return;
   }
   SendVrDeleteCommand(app);
+}
 
-  smart_objects::SmartObject msg_params =
-      smart_objects::SmartObject(smart_objects::SmartType_Map);
+bool DeleteInteractionChoiceSetRequest::Init() {
+  hash_update_mode_ = HashUpdateMode::kDoHashUpdate;
+  return true;
+}
 
-  msg_params[strings::interaction_choice_set_id] = choice_set_id;
-  msg_params[strings::app_id] = app->app_id();
+void DeleteInteractionChoiceSetRequest::on_event(
+    const event_engine::Event& event) {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
 
-  app->RemoveChoiceSet(choice_set_id);
+  if (event.id() == hmi_apis::FunctionID::VR_DeleteCommand) {
+    {
+      sync_primitives::AutoLock auto_lock(requests_lock_);
+      const smart_objects::SmartObject& message = event.smart_object();
+      response_result_code_ = static_cast<hmi_apis::Common_Result::eType>(
```

Comment:
if an error response comes in, but then a successful response comes in, the `response_result_code_` will be success even though an error occurred.

Reply:
@iCollin I replaced this logic with having a vector of result codes. Final result code will be SUCCESS by default, but in case of any error it will contain the first error from this vector